### DISCUSSION TITLE
How to get all contributions via GraphQL API

### DISCUSSION THREAD

**Discussion by nzakas** [opencollective/opencollective#4979] [General]
Hi there,

I'm trying to get all of the contributions in the past month via the GraphQL API. Basically, I'd like to get this list as JSON:
https://opencollective.com/eslint/transactions?type=CREDIT

I can't find a way to do this with the GraphQL API. Here are some of the problems I'm running into:

* Both `Order` and `Transaction` don't have the date of the contribution. `createdAt` and `updatedAt` appear to just show when a monthly order was initiated? I really want the same data as the "datetime" field in the CSV export.
* I'm not getting any one-time donations at all

Here's the query I'm using:

```js
    const query = `{
        account(slug: "eslint") {
          transactions(type: CREDIT, limit: 50) {
            totalCount
            nodes {
              fromAccount {
                name
                website
                imageUrl
              }
              createdAt
              updatedAt
              amount {
                value
              }
            }
          }
        }
      }`;
```

Is it possible to get this information? (It would be a lot easier to play with this was a public GraphiQL UI to use, by the way.)

Thanks!

**Comment by znarf** [comment]
I just tested the query you shared, not sure what's wrong with it:

- The `createdAt` for the `Transaction` looks accurate (time it was recorded) and should be the same as the one in the CSV
- The one for the `Order` is effectively different, it is the time it was first initiated, it doesn't seem to be what you're looking for
- If you add ```order { frequency }```, you should normally see that some `ONETIME` transactions are currently being returned

> It would be a lot easier to play with this was a public GraphiQL UI to use, by the way

I recommend to use something like [GraphQL playground](https://github.com/graphql/graphql-playground) but this is good feedback, we'll look into shipping a live playground.

**Reply by nzakas** [reply]
Okay, I'm not sure what I was doing wrong, but after reading your reply and going back, I see `createdAt` and `updatedAt` now are giving me the values I want. I wonder if I was skipping back and forth between Order and Transaction by mistake. Thanks for your patience!

Unfortunately, GraphQL Playground doesn't work on Windows. :(

By the way, are there any plans to update the GraphQL documentation? There are a lot of fields that don't contain any description, which makes it difficult to understand, for instance, how `createdAt` and `updatedAt` are different, or what they represent on `Transaction` vs. `Order`.

Thanks so much for your help!
